### PR TITLE
fluent-bit: Attempt to unmarshal nested json.

### LINKED ISSUE #4771: fluent-bit Loki output plugin overrides fluent-bit escaped JSON parsing
**Describe the bug**
When using JSON logs with Docker, the JSON will be come escaped and stored in a JSON object like `{"json": "ESCAPED_JSON_STRING"}`. Native fluent-bit seems to handle this (and even has support with [decoders](https://docs.fluentbit.io/manual/pipeline/parsers/decoders) for this exact issue), though when using the loki fluent-bit output plugin, the JSON remains as an escaped string (even when using decoders). 

There seems to be solutions with promtail and Kubernetes for this issue, though I haven't been able to find a solution when using [ECS with firelens](https://grafana.com/docs/loki/latest/clients/aws/ecs/). I ended up forking Loki and creating a new config option `ParseJsonKeys` which works well. I'm happy to open a PR here if that's an acceptable solution, [this](https://github.com/sencha-dev/loki/commit/3972f4b1e3c9dace8079f0a7339fd5277e877e3a) is the relevant commit.

**To Reproduce**
1. Install fluent-bit v1.8.9
2. Checkout loki at tag v2.4.1
3. Run `make fluent-bit-plugin` in the loki repo
4. Create two fluent-bit config files
```
# fluent-bit-native.conf

[INPUT]
    name dummy
    dummy {"log":"{\"level\":\"error\"}","stream":"stdout"}
    samples 1

[Output]
    Name stdout
    Match *
```
```
# fluent-bit-loki.conf

[INPUT]
    name dummy
    dummy {"log":"{\"level\":\"error\"}","stream":"stdout"}
    samples 1

[Output]
    Name grafana-loki
    Match *
    Url ${LOKI_URL}
    LineFormat json
    LogLevel debug
```
5. Run `fluent-bit -e /PATH/TO/LOKI/clients/cmd/fluent-bit/out_grafana_loki.so -c fluent-bit-native.conf`
```
[0] dummy.0: [1637114099.496755000, {"log"=>"{"level":"error"}", "stream"=>"stdout"}]
```
6. Run `fluent-bit -e /PATH/TO/LOKI/clients/cmd/fluent-bit/out_grafana_loki.so -c fluent-bit-loki.conf`
```
level=debug caller=loki.go:53 id=0 msg="processing records" records="map[log:{\"level\":\"error\"} stream:stdout]"
```

**Expected behavior**
For fluent-bit to unescape the JSON string and store it as native JSON instead of a string.

**Environment:**
 - Infrastructure: ECS with firelens, fluent-bit v1.8.9, loki and loki fluent-bit output plugin v2.4.1
 - Deployment tool: Terraform

### PR DESCRIPTION
**Pull request by cyriltovena**
Signed-off-by: Cyril Tovena <cyril.tovena@gmail.com>

**What this PR does / why we need it**:

Attempt to unmarshal for unescaping json content. This is a best effort if it doesn't work the log content is sent with escaping.

**Which issue(s) this PR fixes**:
Fixes #4771

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Review by sandeepsukhani — APPROVED** [review]
LGTM

File: clients/cmd/fluent-bit/loki.go [hunk 1]
```diff
@@ -220,9 +221,21 @@ func removeKeys(records map[string]interface{}, keys []string) {
 	}
 }
 
-func createLine(records map[string]interface{}, f format) (string, error) {
+func (l *loki) createLine(records map[string]interface{}, f format) (string, error) {
 	switch f {
 	case jsonFormat:
+		for k, v := range records {
+			if s, ok := v.(string); ok && strings.Contains(s, "{") {
```
Comment (sandeepsukhani) [inline clients/cmd/fluent-bit/loki.go:228]: What if the value is an array?

Comment (sandeepsukhani) [reply clients/cmd/fluent-bit/loki.go:228]: We could do starts with `{` or `[`. What do you think?

Comment (cyriltovena) [reply clients/cmd/fluent-bit/loki.go:228]: Yeah that sounds fair !

**Pull request merged**